Scripts/datatypes/purpose.py

```diff
@@ -527,14 +528,41 @@ def attempt_calibration(spec: dict):
             # Search deeper
             attempt_calibration(spec[param_name])
 
+def attempt_scaling(spec: dict):
+    """Check if specification includes "calibration" trees.
+
+    Transform regular parameters to include calibration,
+    remove separate calibration parameters.
+    A calibration (named "calibration") tree can be anywhere in the dict,
+    but must refer to existing adjacent parameters or trees of parameters.
+    """
+    try:
+        param_names = list(spec.keys())
+    except AttributeError:
+        # No calibration parameters in this branch, return up
+        return
+    for param_name in param_names:
+        if param_name == "scaling":
+            scale(spec, spec.pop(param_name))
+        else:
+            # Search deeper
+            attempt_scaling(spec[param_name])
+
```

Comment:
Maybe we could skip this separate `attempt_scaling` method and insert this in `attempt_calibration`?
```
        elif param_name == "scaling":
            scale(spec, spec.pop(param_name))
```